velox/functions/sparksql/tests/StringTest.cpp

```diff
@@ -682,5 +682,28 @@ TEST_F(StringTest, translateNonconstantMatch) {
   expected = makeFlatVector<std::string>({"åbaæçè", "åæcèaç"});
   testTranslate({input, match, replace}, expected);
 }
+
+TEST_F(StringTest, urlDecode) {
+  auto UrlDecodeTest = [&](const std::optional<std::string>& input,
+                           const std::string& expected) {
+    EXPECT_EQ(
+        evaluateOnce<std::string>("url_decode(c0)", input).value(), expected);
+  };
+  UrlDecodeTest("AaBbC", "AaBbC");
+  UrlDecodeTest("A+B+C", "A B C");
+  UrlDecodeTest("AaBbC%23", "AaBbC#");
+  UrlDecodeTest("AaBbC%24%5A", "AaBbC$Z");
+  UrlDecodeTest("AaBbC%25FF", "AaBbC%FF");
+  UrlDecodeTest("https%3A%2F%2Fspark.apache.org", "https://spark.apache.org");
+  UrlDecodeTest("https%3A%2F%2Ftts%2Exxx%2Ecom", "https://tts.xxx.com");
+  // Illegal case.
+  EXPECT_THROW(
```

Comment:
Please use VELOX_ASSERT_THROW to check the error message.

Reply:
Yes, just fixed. Thanks!